LoopKitTests/InsulinMathTests.swift

```diff
@@ -346,17 +344,55 @@ class InsulinMathTests: XCTestCase {
             }
         }
     }
+    
+    func testIOBFromDosesWithDifferentInsulinCurves() {
+        let formatter = DateFormatter.descriptionFormatter
+        let f = { (input) in
+            return formatter.date(from: input)!
+        }
+        let output = loadInsulinValueFixture("iob_from_multiple_curves_output")
+
+        let fiasp = ExponentialInsulinModelPreset.fiasp
+        let child = ExponentialInsulinModelPreset.humalogNovologChild
+
+        let doses = [
+            DoseEntry(type: .basal, startDate: f("2018-05-15 14:42:36 +0000"), endDate: f("2018-05-16 14:42:36 +0000"), value: 0.84999999999999998, unit: .unitsPerHour, syncIdentifier: "7b02646a070f120e2200", scheduledBasalRate: nil),
+            DoseEntry(type: .bolus, startDate: f("2018-05-15 14:44:46 +0000"), endDate: f("2018-05-15 14:44:46 +0000"), value: 0.9, unit: .units, syncIdentifier: "01004a004a006d006e22354312", scheduledBasalRate: nil),
+            DoseEntry(type: .tempBasal, startDate: f("2018-05-15 14:42:36 +0000"), endDate: f("2018-05-15 14:42:36 +0000"), value: 0.0, unit: .unitsPerHour, syncIdentifier: "1600646a074f12", scheduledBasalRate: nil),
+            DoseEntry(type: .tempBasal, startDate: f("2018-05-15 14:32:51 +0000"), endDate: f("2018-05-15 15:02:51 +0000"), value: 1.8999999999999999, unit: .unitsPerHour, syncIdentifier: "16017360074f12", scheduledBasalRate: nil),
+            DoseEntry(type: .bolus, startDate: f("2018-05-15 14:52:51 +0000"), endDate: f("2018-05-15 15:52:51 +0000"), value: 0.9, unit: .units, syncIdentifier: "01004a004a006d006e22354312", scheduledBasalRate: nil),
+        ]
+
+        let iobWithoutModel = doses.insulinOnBoard(defaultModel: exponentialModel, longestEffectDuration: exponentialModel.effectDuration)
+
+        let dosesWithModel = [
```

Comment:
Unit test that IOB accounting is correct when there doses with different insulin model curves